app/helpers/google_analytics_helper.rb

```diff
@@ -9,12 +9,34 @@ module GoogleAnalyticsHelper
   # fields tracked by Google.
   # https://developers.google.com/analytics/devguides/collection/gtagjs/setting-values
 
+  def ga_user_properties
+    retval = {}
+    unless ( value = ga_collection_id ).blank?
+      retval['collection_id'] = ":#{value}:"
+    end
+    unless ( value = ga_page_type ).blank?
+      retval['page_type'] = value
+    end
+    unless ( value = ga_repository_id ).blank?
+      retval['repository_id'] = ":#{value}:"
+    end
+    retval.to_json.html_safe
+  end
+
   def ga_collection_id
     # within_collection_context? doesn't quite help b/c we can't get the slug
     # out of a within-collection search, just the title
     @document&.eadid
   end
 
+  def ga_repository_id
+    if params[:f].present? and params[:f]["repository_sim"].present?
```

Comment:
Should use `&&` over `and` here. Hi, Larry!

Reply:
Weird --- pretty sure I mentally ported the Python `and` to Ruby due to the similaritiy between the two. Perl/PHP use `&&`. 🤯